### PR TITLE
Fix some problems in UI

### PR DESCRIPTION
**Pull request by besser82**
* Fix crash when shutting down the UI (fixes https://github.com/libyui/libyui-qt/issues/41)
* Fix a problem with hanging UI
* new Version: 2.47.1

### CODE REVIEW

File: src/YQReplacePoint.cc [hunk 1]
```diff
@@ -44,8 +44,16 @@ void YQReplacePoint::showChild()
 
     if ( child )
     {
-	QWidget * qChild = (QWidget *) child->widgetRep();
-	qChild->show();
+        QWidget * qChild = (QWidget *) child->widgetRep();
```
Comment (jreidinger) [inline src/YQReplacePoint.cc]: in general fix looks ok, but please write here comment why such complex stuff is needed to be done as it is not obvious. You do not need to comment what you do, only why it is needed to be done this way.

Comment (besser82) [inline src/YQReplacePoint.cc]: Added comments in rebased commit

File: src/YQReplacePoint.cc [hunk 2]
```diff
@@ -42,10 +42,21 @@ void YQReplacePoint::showChild()
...
 
+    /* This needs to be done so the UI won't crash if the are no children. */
     if ( child )
     {
-	QWidget * qChild = (QWidget *) child->widgetRep();
-	qChild->show();
+        QWidget * qChild = (QWidget *) child->widgetRep();
+        qChild->show();
+
+        /* Refresh the whole UI so it doesn't get stalled in some situations. */
```
Comment (jreidinger) [inline src/YQReplacePoint.cc]: great, much better now, just one final note, can you add also reference to github issue, so we have in future place where we can see some discussion about this issue? Is there such issue?

File: package/libyui-qt.changes [hunk 3]
```diff
@@ -1,4 +1,11 @@
 -------------------------------------------------------------------
+Sun Feb  5 14:43:49 CET 2017 - besser82@fedoraproject.org
+
+- Fix crash when shutting down the UI
```
Comment (jreidinger) [inline package/libyui-qt.changes]: please reference here bug reports like `(gh#libyui-libyui#41, bsc#931762)` so we have reference what problems we are fixing. Thanks and same for hanging if such issue exists.

File: src/YQReplacePoint.cc (hunk 2 above)
Comment (jreidinger) [inline src/YQReplacePoint.cc]: and if such issue do not exists, is there some email discussion about this or something similar?

Comment (besser82) [inline src/YQReplacePoint.cc]: For this particulary problem there is no open issue.

File: package/libyui-qt.changes (hunk 3 above)
Comment (besser82) [inline package/libyui-qt.changes]: Added.

File: package/libyui-qt.changes [hunk 4]
```diff
@@ -1,4 +1,11 @@
 -------------------------------------------------------------------
+Sun Feb  5 14:43:49 CET 2017 - besser82@fedoraproject.org
+
+- Fix crash when shutting down the UI (gh#libyui-libyui-qt#41)
```
Comment (jreidinger) [inline package/libyui-qt.changes]: thanks, please add also reference to bugzilla, so we can link it and auto-submit it to suse. Thanks ( just add there `, bsc#931762)` then I will merge it.

Comment (besser82) [inline package/libyui-qt.changes]: Done.  Thank you!

**Comment by jreidinger** [comment]
thanks

**Pull request merged**